### ISSUE TITLE
Bitmap smoothing

### ISSUE THREAD

**Issue by MiltonW** [Beeblerox/BitmapFont#19]
Would be nice to have an option to set pixelsnapping and smoothing:

```
    _bitmap = new Bitmap(_bitmapData, PixelSnapping.AUTO, true));           
```

**Comment by Beeblerox** [comment]
@MiltonW what about my latest push? does it work as you prefer?

**Comment by MiltonW** [comment]
I think so, but I seem to have a problem with Flash (and HTML5).

Linux/Neko target:
![bla](https://cloud.githubusercontent.com/assets/6236680/4518094/331899e4-4c72-11e4-856a-7728d4b15a69.png)

Flash:
![bla2](https://cloud.githubusercontent.com/assets/6236680/4518095/387dc184-4c72-11e4-86ad-e09a906737d1.png)

Same code, no Flash conditionals.
This occurs when I have

```
     text_retro.size = 0.57;                                             
```

If I don't use that (or use a nice round integer) it looks just as good as the other targets.
But if I then try to use text_retro.scaleX = 0.57 I get:

```
Invalid field access : set_scaleX
```

So I can't use the DisplayObject scaling anymore, which I would guess would not distort the image.

**Comment by Beeblerox** [comment]
let me think about it

**Comment by Beeblerox** [comment]
@MiltonW sorry that i haven't worked on it earlier, but i think i've fixed this issue.
could you try updated code?

**Comment by MiltonW** [comment]
No change I think. When I use HTML5/Flash I have the following behaviour:

```
     text_retro.size = 0.57;
```

![size](https://cloud.githubusercontent.com/assets/6236680/4589586/b29747b0-5055-11e4-8979-876f49fdc036.png)

```
     text_retro.scaleX = 0.57; text_retro.scaleY = 0.57;
```

![scale](https://cloud.githubusercontent.com/assets/6236680/4589591/b99f3fcc-5055-11e4-8230-114f7ff2db0c.png)

Since scaleX/Y seems to work it is not really a problem (for me anyway :) ). Can't reproduce the 'Invalid field access : set_scaleX' anymore.

PS. Does look a bit smoother now that I look at it, but I don't think it has to do with smoothing...

**Comment by Beeblerox** [comment]
strange. this is what i've got on flash target with latest code (with size set to 0.57):
smoothing off
![smoothing_off](https://cloud.githubusercontent.com/assets/857406/4590490/330a9334-5060-11e4-877c-97fee4b54bec.png)
smoothing on
![smoothing_on](https://cloud.githubusercontent.com/assets/857406/4590492/3b66e10e-5060-11e4-9222-ff048401d657.png)

**Comment by MiltonW** [comment]
Yeah, a bit smoother, but what does it look like without size, and then scale it with scaleX/Y?

**Comment by Beeblerox** [comment]
it looks the same

**Comment by Beeblerox** [comment]
ok, here is another one fix.
it looks like if you change Bitmap's bitmapData then smoothing isn't applying to it anymore. so we have to reapply smoothing

**Comment by MiltonW** [comment]
It all doesn't seem to make a difference. All the smoothing does is smooth the jagged edges. The edges aren't there if I don't use size. So then I can use scaleX/Y and it stays crisp.

Try going fullscreen or even bigger and try to keep it this clean, no edges at all:

![scale](https://cloud.githubusercontent.com/assets/6236680/4590987/795e0ce8-5066-11e4-85ce-512dab77efeb.png)

Again, if I then comment scaling, and use size:

![size](https://cloud.githubusercontent.com/assets/6236680/4590997/a3485bb2-5066-11e4-8c7b-1196a8d6ab39.png)

This is only on Flash/HTML5. Using size = 0.57 and scaling the entire stage by 3.

**Comment by Beeblerox** [comment]
oh, so you were scaling stage also. and i've tested without it.
I guess it will be impossible to achieve this by using size then

**Comment by MiltonW** [comment]
No, I'm not scaling the stage, that was just to make the difference more obvious.

**Comment by FlashTang** [comment]
Hi there ,@Beeblerox  Now how to disable smoohting ? 
I'm using this frok by @flowplay
 https://github.com/flowplay/BitmapFont
angelCodeField.smoothing = false; // not working , it's always smoothing, I'm using pxiel font , so need to be non-smoothing
p.s. when set scale to 2 , it's smoothing with white blurry border
Thanks

**Comment by FlashTang** [comment]
Just find a way to fix :
Find this line 
`_tilemap = new Tilemap(tileMapWidth, tileMapHeight, font.tileset);`
Add this.smoothing to the end 
`_tilemap = new Tilemap(tileMapWidth, tileMapHeight, font.tileset,this.smoothing);`
